src/extractor/guidance/intersection.cpp

```diff
@@ -112,6 +117,45 @@ Intersection::getHighestConnectedLaneCount(const util::NodeBasedDynamicGraph &gr
     const auto view = *this | boost::adaptors::transformed(to_lane_count);
     boost::range::find_if(view, extract_maximal_value);
     return max_lanes;
+
+    return max_lanes;
+}
+
+// Intersections are sorted roads: [0] being the UTurn road, then from sharp right to sharp left.
+
+ConnectedRoad &Intersection::getUTurnRoad()
+{
+    BOOST_ASSERT(valid());
```

Comment:
while technically reasonable, checking the validity at every such iterator could have a serious performance impact on debug builds/release with assertion builds.

I would say that we should rely on the generators to assert validity to not run into problems with debug builds/assertion builds on the demo server.